tdrs-frontend/src/components/FileUpload/FileUpload.jsx

```diff
@@ -89,6 +93,25 @@ function FileUpload({ section, setLocalAlertState }) {
     filereader.onloadend = (evt) => {
       /* istanbul ignore next */
       if (!evt.target.error) {
+        // Validate file extension before proceeding
+        const re = /(\.txt|\.ms\d{2}|\.ts\d{2,3})/i
```

Comment:
Nit: Why not settings a regex that checks the ext only. What I mean is you shouldn't need to define ext below.